src/mock/quick_p2p/network.rs

```diff
@@ -218,13 +216,31 @@ impl Queue {
         self.0.push_back(packet)
     }
 
-    fn pop(&mut self) -> Option<Packet> {
-        self.0.pop_front()
-    }
-
     fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
+
+    // This function will pop random msg from the queue.
+    fn pop_random_msg(&mut self, rng: &mut SeededRng) -> Option<Packet> {
+        let first_non_random_msg = self
+            .0
+            .iter()
+            .position(|packet| {
+                if let Packet::Disconnect = packet {
+                    true
+                } else {
+                    false
+                }
+            })
+            .unwrap_or(0);
+
+        let selected = if first_non_random_msg == 0 {
+            0
+        } else {
+            rng.gen_range(0, first_non_random_msg)
+        };
```

Comment:
This can be more succinctly expressed as `let selected = rng.gen_range(0, cmp::max(first_non_random_msg, 1))`